assets/javascripts/discourse/components/modal/edit-topic-assignments.js

```diff
@@ -0,0 +1,46 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import I18n from "I18n";
+
+export default class EditTopicAssignments extends Component {
+  @service taskActions;
+  @tracked assignments = [];
+
+  constructor() {
+    super(...arguments);
+    this.assignments = this.topic.assignments();
+  }
+
+  get title() {
+    if (this.topic.isAssigned() || this.topic.hasAssignedPosts()) {
+      return I18n.t("edit_assignments_modal.title");
+    } else {
+      return I18n.t("discourse_assign.assign_modal.title");
+    }
+  }
+
+  get topic() {
+    return this.args.model.topic;
+  }
+
+  @action
+  async submit() {
+    this.args.closeModal();
+    try {
+      await this.#assign();
+    } catch (error) {
+      popupAjaxError(error);
+    }
+  }
+
+  async #assign() {
+    for (const assignment of this.assignments) {
+      if (assignment.isEdited) {
```

Comment:
We don't send server requests for assignments that have not been edited. In case user edits several assignments, it will send several requests to the server. I'm not sure if it's worth introducing a server method here to do all that in one request, but it may be done in a separate PR in the future if case we decide we want it.